src/hera/workflows/script.py

```diff
@@ -578,11 +567,10 @@ def _extract_all_output_annotations(source: Callable) -> List:
     output = []
 
     for _, func_param in inspect.signature(source).parameters.items():
-        if get_origin(func_param.annotation) is Annotated:
-            annotation_args = get_args(func_param.annotation)
-            annotated_type = annotation_args[1]
-            if isinstance(annotated_type, (Artifact, Parameter)) and annotated_type.output:
-                output.append(annotation_args)
+        if (
+            annotated := _type_util.get_annotated_metadata(func_param.annotation, (Artifact, Parameter))
+        ) and annotated.output:
+            output.append(get_args(func_param.annotation))
```

Comment:
I think this was technically a bug in the old code (but the actual list of value is never used so it didn't surface) and this should be
```suggestion
            output.append(annotated)
```
Which also saves another call :)